common/flatpak-installation.c

```diff
@@ -1112,9 +1115,16 @@ flatpak_installation_list_installed_refs_for_update (FlatpakInstallation *self,
        * This makes sure that the ref (maybe an app or runtime) remains in usable
        * state and fixes itself through an update.
        */
-      state = flatpak_dir_get_remote_state_optional (dir, remote_name, FALSE, cancellable, error);
+      state = g_hash_table_lookup (remote_states, remote_name);
       if (state == NULL)
-        continue;
+        {
+
+          state = flatpak_dir_get_remote_state_optional (dir, remote_name, FALSE, cancellable, error);
```

Comment:
We need to pass `NULL` instead of `error` here. Otherwise, we risk re-using an already set `GError` pointer. This bug existed before your patch, but still worth fixing.

Reply:
Submitted as https://github.com/flatpak/flatpak/pull/3280

Reply:
That has been merged upstream, so it probably makes sense to backport it downstream in this PR too.

Reply:
LGTM other than that

Reply:
Is this still relevant? Seems to have stalled.